.github/workflows/build_windows.yml

```diff
@@ -321,58 +270,54 @@ jobs:
-        echo "DO_BUILD: $DO_BUILD"
-        echo "DO_UPDATE_TS=$DO_UPDATE_TS" >> $GITHUB_ENV
-        echo "DO_UPDATE_TS: $DO_UPDATE_TS"
-        echo "DO_PUBLISH=$DO_PUBLISH" >> $GITHUB_ENV
-        echo "DO_PUBLISH: $DO_PUBLISH"
-        echo "UPLOAD_ARTIFACT_NAME=$UPLOAD_ARTIFACT_NAME" >> $GITHUB_ENV
-        echo "UPLOAD_ARTIFACT_NAME: $UPLOAD_ARTIFACT_NAME"
+        echo "BUILD_MODE=$BUILD_MODE" | tee -a $GITHUB_ENV
+        echo "BUILD_NUMBER=$BUILD_NUMBER" | tee -a $GITHUB_ENV
+        echo "DO_UPDATE_TS=$DO_UPDATE_TS" | tee -a $GITHUB_ENV
+        echo "DO_PLACEHOLDER_TRANSLATIONS=$DO_PLACEHOLDER_TRANSLATIONS" | tee -a $GITHUB_ENV
+        echo "DO_PUBLISH=$DO_PUBLISH" | tee -a $GITHUB_ENV
+        echo "UPLOAD_ARTIFACT_NAME=$UPLOAD_ARTIFACT_NAME" | tee -a $GITHUB_ENV
 
     - name: Setup environment
-      if: env.DO_BUILD == 'true'
       run: |
         buildscripts\ci\windows\setup.bat --portable ON
     - name: Make environment file
-      if: env.DO_BUILD == 'true'
       shell: bash
       run: |
         bash ./buildscripts/ci/windows/make_environment.sh
+    - name: Generate _en.ts files
+      shell: bash
+      env:
+        LUPDATE_ARGS: ""
+        POSTPROCESS_ARGS: "--warn-only ${{ env.DO_PLACEHOLDER_TRANSLATIONS == 'true' && '--generate-placeholder-translations' || '' }}"
+        POSTPROCESS_LAUNCHER: "python3 -X utf8"
+      run: |
+        bash ./buildscripts/ci/translation/run_lupdate.sh
     - name: Update .ts files
       if: env.DO_UPDATE_TS == 'true'
       shell: bash
       run: |
         bash ./buildscripts/ci/translation/tx_install.sh -t ${{ secrets.TRANSIFEX_API_TOKEN }} -s windows
         bash ./buildscripts/ci/translation/tx_pull.sh
     - name: Build
-      if: env.DO_BUILD == 'true'
       shell: cmd
       run: |
         IF ${{ env.SENTRY_URL != 0 }} == true ( SET C_URL="${{ env.SENTRY_URL }}" ) ELSE ( SET C_URL="" )
         buildscripts\ci\windows\build.bat --portable ON -n ${{ env.BUILD_NUMBER }} --crash_log_url %C_URL%
     - name: Package
-      if: env.DO_BUILD == 'true'
       shell: cmd
-      run: |      
+      run: |
         IF ${{ secrets.WIN_SIGN_CERTIFICATE_ENCRYPT_SECRET != 0 }} == true ( SET S_S=${{ secrets.WIN_SIGN_CERTIFICATE_ENCRYPT_SECRET }} ) ELSE ( SET S_S="''" )
         IF ${{ secrets.WIN_SIGN_CERTIFICATE_PASSWORD != 0 }} == true ( SET S_P=${{ secrets.WIN_SIGN_CERTIFICATE_PASSWORD }} ) ELSE ( SET S_P="''" )
         buildscripts\ci\windows\package.bat --portable ON --signsecret %S_S% --signpass %S_P%
-    - name: Checksum 
-      if: env.DO_BUILD == 'true'
+    - name: Checksum
       run: |
-        bash ./buildscripts/ci/tools/checksum.sh     
-    - name: Publish package
+        bash ./buildscripts/ci/tools/checksum.sh
+    - name: Publish to OSUOSL
       if: env.DO_PUBLISH == 'true'
       shell: bash
       run: |
         bash ./buildscripts/ci/tools/osuosl/publish.sh -s ${{ secrets.OSUOSL_SSH_ENCRYPT_SECRET }} --os windows -v 4 --arch x86_64-portable
```

Comment:
The `latest` symlink for [Windows nightlies](https://ftp.osuosl.org/pub/musescore-nightlies/windows/4x/nightly/?C=M;O=D) is called `portable.exe` rather than `paf.exe`. I think this line is the culprit.

I couldn't find it earlier because it didn't show it my my search of the source code. That must be because it's in the hidden `.github` folder.

Reply:
It looks like one more step was necessary for that, see the changes of the last commit in `publish.sh`. Not super proud of it; if you have a more elegant solution that would be great.

Reply:
Technically these would work and avoid spawning a subshell:

```Bash
ext="${file/*.paf.exe/paf.exe}"
ext="${ext#"${file%.*}."}"
```
```Bash
ext="${file%paf.exe}"
ext="${file#${ext}}"
ext="${ext:-${file##*.}}"
```
But I think your solution is more readable.